### DISCUSSION TITLE
JIT when content not in files (e.g. builder tool)

### DISCUSSION THREAD

**Discussion by tslocke** [tailwindlabs/tailwindcss#8944] [Help]
The Tailwind JIT seems to be built on the assumption that the content is in the file system.

In a site builder tool this might not be the case - someone is editing the site and adding css classes at runtime.

For this kind of situation one would need to access tailwind as a library and invoke the jit as needed, passing in the content that needs to be scanned. 

I've seen the advice that tailwind can be used programatically by using PostCSS programatically, but I think tailwind is still going to look to the filesystem to find the content.

This guy figured it out: https://github.com/beyondcode/tailwindcss-jit-cdn

But that uses some pretty crazy magic to provide a faked version of the `fs` module to tailwind. I was wondering if there might be an easier way : )

**Comment by tslocke** [comment]
Found this : )

https://tailwindcss.com/docs/content-configuration#configuring-raw-content

**Comment by arisac** [comment]
sounds like how Play CDN works > https://tailwindcss.com/docs/installation/play-cdn, but it's not recommended for production to use, just dev use only.

the real time edit feature on https://play.tailwindcss.com/ is maybe what you want for site editing. it's source code at https://github.com/tailwindlabs/play.tailwindcss.com
